src/styles/main.scss

```diff
@@ -0,0 +1,65 @@
+$daySize: 100px;
+$countDays: 31;
+$gap: 1px;
+$calendarWidth: $daySize * 7 + 10;
+$daysOfWeek: ('mon' 1, 'tue' 2, 'wed' 3, 'thu' 4, 'fri' 5, 'sat' 6, 'sun' 7);
+
+body {
+  display: flex;
+  align-items: center;
+  justify-content: center;
+  height: 100vh;
+  margin: 0;
+  font-family: Arial, Helvetica, sans-serif;
+  font-size: 30px;
+}
+
+.calendar {
+  flex-wrap: wrap;
+  display: flex;
+  margin-left: 4px;
+  padding: 10px;
+  gap: $gap;
+  max-width: $calendarWidth;
+
+  &__day {
+    display: flex;
+    box-sizing: border-box;
+    justify-content: center;
+    align-items: center;
+    background-color: #eee;
+    border: 1px solid black;
+    height: $daySize;
+    width: $daySize;
+    transition-duration: 0.5s;
+  }
+
+  &__day:hover {
+    cursor: pointer;
+    background-color: #ffbfcb;
+    transform: translateY(-20px);
+    animation-duration: 500ms;
+  }
+
+  @for $day from 1 through 31 {
+    .calendar__day:nth-child(#{$day})::before {
+      content: '#{$day}';
+    }
+  }
+
+  @for $deDay from 28 to 31 {
+    .calendar--month-day-length-#{$deDay} {
+      @for $de from ($deDay + 1) through 31 {
+        & :nth-child(n + #{$de}) {
+          display: none;
+        }
+      }
+    }
+  }
+
+  @each $name, $value in $daysOfWeek {
+    .calendar--start-day-#{$name} .calendar__day:first-child {
+      margin-left: ($value * ($daySize + $gap)) - ($daySize + $gapValue);
```

Comment:
There is a typo in the variable name `$gapValue`. It should be `$gap` to match the variable defined at the beginning of the file.